apps/dashboard/lib/trpc/routers/key/create.ts

```diff
@@ -100,7 +103,22 @@ export const createKey = t.procedure
             "We are unable to create the key. Please contact support using support.unkey.dev",
         });
       });
-
+    if (input.recoverEnabled && keyAuth?.storeEncryptedKeys) {
+      const vault = new Vault(env().AGENT_URL, env().AGENT_TOKEN);
+      const encryptReq: EncryptRequest = {
+        keyring: workspace.id,
+        data: key,
+      };
+      const requestId = crypto.randomUUID();
+      const context: RequestContext = { requestId };
+      const vaultRes = await vault.encrypt(context, encryptReq);
```

Comment:
What happens in vault creation fails here? You aren't handling the negative path:

For example encryption fails but the user receives no error or success message, The key gets upserted into the database but without encryption, on the frontend to inform them that it failed. 

```
[
    {
        "error": {
            "json": {
                "message": "Unable to encrypt, fetch error: error code: 1016",
                "code": -32603,
                "data": {
                    "code": "INTERNAL_SERVER_ERROR",
                    "httpStatus": 500,
                    "stack": "Error: Unable to encrypt, fetch error: error code: 1016\n    at (vc/edge/function:298:40314)\n    at (vc/edge/function:298:43255)\n    at (vc/edge/function:295:126266)\n    at (vc/edge/function:295:125662)\n    at (vc/edge/function:295:125662)\n    at (vc/edge/function:295:125662)\n    at (vc/edge/function:295:125985)\n    at (vc/edge/function:337:12967)\n    at (vc/edge/function:337:14421)",
                    "path": "key.create"
                }
            }
        }
    }
]
```

Reply:
![image](https://github.com/user-attachments/assets/5e2ed8ff-c49c-407f-b2ec-91faaa541e01)
New changes will display the toast like this if the encryption fails